src/Publisher/FilesystemPublisher.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+namespace SilverStripe\StaticPublishQueue\Publisher;
+
+use SilverStripe\Assets\Filesystem;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\StaticPublishQueue\Publisher;
+use function SilverStripe\StaticPublishQueue\URLtoPath;
+
+class FilesystemPublisher extends Publisher
+{
+    /**
+     * @var string
+     */
+    protected $destFolder = 'cache';
+
+    /**
+     * @var string
+     */
+    protected $fileExtension = 'php';
+
+    /**
+     * @return string
+     */
+    public function getDestPath()
+    {
+        $base = defined('PUBLIC_PATH') ? PUBLIC_PATH : BASE_PATH;
+        return $base . DIRECTORY_SEPARATOR . $this->getDestFolder();
+    }
+
+    public function setDestFolder($destFolder)
+    {
+        $this->destFolder = $destFolder;
+        return $this;
+    }
+
+    public function getDestFolder()
+    {
+        return $this->destFolder;
+    }
+
+    public function setFileExtension($fileExtension)
+    {
+        $fileExtension = strtolower($fileExtension);
+        if (!in_array($fileExtension, ['html', 'php'])) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'Bad file extension "%s" passed to %s::%s',
+                    $fileExtension,
+                    static::class,
+                    __FUNCTION__
+                )
+            );
+        }
+        $this->fileExtension = $fileExtension;
+        return $this;
+    }
+
+    public function getFileExtension()
+    {
+        return $this->fileExtension;
+    }
+
+    public function purgeURL($url)
+    {
+        if (!$url) {
+            user_error("Bad url:" . var_export($url, true), E_USER_WARNING);
+            return;
+        }
+        if ($path = $this->URLtoPath($url)) {
+            $success = $this->deleteFromPath($path . '.html') && $this->deleteFromPath($path . '.php');
+            return [
+                'success' => $success,
+                'url' => $url,
+                'path' => $this->getDestPath() . DIRECTORY_SEPARATOR . $path,
+            ];
+        }
+        return [
+            'success' => false,
+            'url' => $url,
+            'path' => false,
+        ];
+    }
+
+    /**
+     * @param string $url
+     * @return array A result array
+     */
+    public function publishURL($url, $forcePublish = false)
```

Comment:
Up for debate, but I'd rather see a PSR-2 compatible `publishUrl()` method name than a semantically correct `publishURL()` method name

Reply:
Our coding convention has been to capitalise abbreviations for quite some time.

PSR-2 gives no opinion on this from what I can read: https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md